ironfish/src/evm/evm.ts

```diff
@@ -183,7 +190,18 @@ export type EvmUnshield = {
 
 export type UTXOEvent = EvmShield | EvmUnshield
 
-export type EvmResult = {
-  result: RunTxResult
-  events: UTXOEvent[]
+export type EvmResult =
+  | {
+      result: RunTxResult
+      error: undefined
+      events: UTXOEvent[]
+    }
+  | {
+      result: undefined
+      error: EvmError
+      events: UTXOEvent[]
```

Comment:
should events be undefined here?

Reply:
Yeah I think that it's probably more correct for it to be undefined